main.go

```diff
@@ -20,9 +21,31 @@ func main() {
 	os.Exit(code)
 }
 
+func subscribeToSignals(sig os.Signal) <-chan os.Signal {
+	interrupts := make(chan os.Signal, 1)
+	signal.Notify(interrupts, sig)
+	return interrupts
+}
+
 /* Command parsing */
 
-type CliCommandParser struct{}
+func NewCliCommandParser(interrupts <-chan os.Signal) *CliCommandParser {
+	return &CliCommandParser{
+		Interrupts: interrupts,
+		NewCommandToRunHTTPServer: func(contentRootPath string, host string, port uint16) (CliCommand, chan bool) {
```

Comment:
I might create a named function for this - seems like a lot of logic to declare anonymously

Reply:
Yeah, it's kind of a mouthful.  Let me rebase and see what I can do.